core/ir/opnd_api.h

```diff
@@ -2009,6 +2008,18 @@ DR_API
 opnd_t
 opnd_create_immed_int(ptr_int_t i, opnd_size_t data_size);
 
+#ifdef RISCV64
+DR_API
+/**
+ * Returns a signed immediate integer operand with value \p i and size
+ * \p data_size; \p data_size must be a OPSZ_ constant.
+ *
+ * The integer will be formatted as a decimal when disassemble.
+ */
+opnd_t
+opnd_create_immed_int_decimal(ptr_int_t i, opnd_size_t data_size);
```

Comment:
How about removing this function and instead using opnd_add_flags(onpd_create_immed_int(), DR_OPND_IMM_PRINT_DECIMAL).  That is what opnd_add_flags is for, to compose so we don't need to make special routines for combinations.

Reply:
Yes, it's better, done in https://github.com/DynamoRIO/dynamorio/pull/6095/commits/f6aacbe145f8f671760d4f1639f4d0994937b3ea.